src/components/NftGallery/NftGallery.tsx

```diff
@@ -9,49 +9,65 @@ import { Collection } from 'types/Collection';
 import { Nft } from 'types/Nft';
 import { insertWhitespaceBlocks } from 'utils/collectionLayout';
 import { WhitespaceBlock } from 'flows/shared/steps/OrganizeCollection/types';
+import { useFragment } from 'react-relay';
+import { graphql } from 'relay-runtime';
+import { NftGalleryFragment$key } from '__generated__/NftGalleryFragment.graphql';
+import { useCollectionColumns } from 'hooks/useCollectionColumns';
+import { removeNullValues } from 'utils/removeNullValues';
 
 type Props = {
-  collection: Collection;
+  collectionRef: NftGalleryFragment$key;
   mobileLayout: DisplayLayout;
 };
 
-function isNft(item: Nft | WhitespaceBlock): item is Nft {
-  return 'created_at' in item;
-}
+function NftGallery({ collectionRef, mobileLayout }: Props) {
+  const collection = useFragment(
+    graphql`
+      fragment NftGalleryFragment on GalleryCollection {
+        dbid
+        layout {
+          columns
+          whitespace
+        }
+        nfts {
+          id
+          ...NftPreviewFragment
+        }
 
-function NftGallery({ collection, mobileLayout }: Props) {
-  const columns = useMemo(() => {
-    if (collection?.layout?.columns && isValidColumns(collection.layout.columns)) {
-      return collection.layout.columns;
-    }
+        ...useCollectionColumnsFragment
+      }
+    `,
+    collectionRef
+  );
 
-    return DEFAULT_COLUMNS;
-  }, [collection.layout]);
+  const columns = useCollectionColumns(collection);
 
   const hideWhitespace = mobileLayout === DisplayLayout.LIST;
+  const nonNullWhitespace = removeNullValues(collection.layout?.whitespace);
 
   const collectionWithWhitespace = useMemo(
-    () => insertWhitespaceBlocks(collection.nfts, collection.layout?.whitespace ?? []),
+    () => insertWhitespaceBlocks(collection.nfts ?? [], nonNullWhitespace),
     [collection.layout?.whitespace, collection.nfts]
   );
+
   const itemsToDisplay = useMemo(
     () => (hideWhitespace ? collection.nfts : collectionWithWhitespace),
     [collection.nfts, collectionWithWhitespace, hideWhitespace]
   );
 
   return (
     <StyledCollectionNfts columns={columns} mobileLayout={mobileLayout}>
-      {itemsToDisplay.map((item) =>
-        isNft(item) ? (
-          // temporarily fix bugged NFTs from not appearing within a collection.
-          // in the future, the backend will simply not return them.
-          Boolean(item.id) && (
-            <NftPreview key={item.id} nft={item} collectionId={collection.id} columns={columns} />
-          )
```

Comment:
@Robinnnnn Let's talk this through and make sure the GraphQL stuff doesn't cause any regressions.

Reply:
all good for now, this was caused by another bug that we fixed 👍